pkg/setting/setting.go

```diff
@@ -802,19 +803,18 @@ func (cfg *Cfg) loadSpecifiedConfigFile(configFile string, masterFile *ini.File)
 	return nil
 }
 
-func (cfg *Cfg) loadConfiguration(args CommandLineArgs) (*ini.File, error) {
+func (cfg *Cfg) loadConfiguration(args CommandLineArgs) (parsedFile *ini.File, err error) {
 	// load config defaults
 	defaultConfigFile := path.Join(HomePath, "conf/defaults.ini")
 	configFiles = append(configFiles, defaultConfigFile)
 
-	// check if config file exists
-	if _, err := os.Stat(defaultConfigFile); os.IsNotExist(err) {
-		fmt.Println("Grafana-server Init Failed: Could not find config defaults, make sure homepath command line parameter is set or working directory is homepath")
-		os.Exit(1)
+	// load config file if exists; use embedded defaults if does not exist
+	if _, err = os.Stat(defaultConfigFile); os.IsNotExist(err) {
+		parsedFile, err = ini.Load(grafana.DefaultsINI)
```

Comment:
I think the purpose of this pull request is generally a great idea, because I totally agree that `Using embedded defaults is still better than exiting with an error`, what I'm not sure about is whether returning the error (if so) from reading the embedded defaults is a good idea, especially considering that the printed error still looks like: `Failed to parse defaults.ini`, while there won't be any (`defaults.ini` file).

WDYT? 🤔

Reply:
agreed, i think that message should be changed for the embedded case. By embedding the defaults.ini, we effectively make parseability of the embedded `defaults.ini` a build-time consideration, rather than runtime.

That makes the audience for such an error message different (GL devs, not users), and it becomes a problem that should never escape CI.